Non-overlapping identity check false-positive starting in 0.780

https://mypy-play.net/?mypy=latest&python=3.8&flags=strict&gist=9b082c260ecb9586b358a06c8583edc2

```
class A:
    y: bool
    def update(self) -> None:
        pass
    
a = A()
assert a.y is True
a.update()
assert a.y is False
```

gives

```
main.py:9: error: Non-overlapping identity check (left operand type: "Literal[True]", right operand type: "Literal[False]")
Found 1 error in 1 file (checked 1 source file)
```

This may be reasonable opinionated behavior, but wanted to flag it as a behavior change between 0.770 and 0.780
I would personally consider this to be a false positive. It's not obvious how to change the source code to make mypy happy here.

--- a/mypy/checkexpr.py
+++ b/mypy/checkexpr.py
@@ -2306,6 +2306,10 @@ class ExpressionChecker(ExpressionVisitor[Type]):
                 left = map_instance_to_supertype(left, abstract_set)
                 right = map_instance_to_supertype(right, abstract_set)
                 return not is_overlapping_types(left.args[0], right.args[0])
+        if isinstance(left, LiteralType) and isinstance(right, LiteralType):
+            if isinstance(left.value, bool) and isinstance(right.value, bool):
+                # Comparing different booleans is not dangerous.
+                return False
         return not is_overlapping_types(left, right, ignore_promotions=False)
 
     def get_operator_method(self, op: str) -> str:

--- a/test-data/unit/check-literal.test
+++ b/test-data/unit/check-literal.test
@@ -3228,3 +3228,18 @@ y: Literal[F.A]
 reveal_type(x)  # N: Revealed type is 'Literal[__main__.Foo.A]'
 reveal_type(y)  # N: Revealed type is 'Literal[__main__.Foo.A]'
 [builtins fixtures/tuple.pyi]
+
+[case testStrictEqualityLiteralTrueVsFalse]
+# mypy: strict-equality
+
+class C:
+    a = True
+
+    def update(self) -> None:
+        self.a = False
+
+c = C()
+assert c.a is True
+c.update()
+assert c.a is False
+[builtins fixtures/bool.pyi]

